addons/maintenance/models/maintenance.py

```diff
@@ -350,45 +344,12 @@ def write(self, vals):
         res = super(MaintenanceRequest, self).write(vals)
         if vals.get('owner_user_id') or vals.get('user_id'):
             self._add_followers()
-        if 'stage_id' in vals:
-            self.filtered(lambda m: m.stage_id.done).write({'close_date': fields.Date.today()})
-            self.filtered(lambda m: not m.stage_id.done).write({'close_date': False})
-            self.activity_feedback(['maintenance.mail_act_maintenance_request'])
-            self.activity_update()
-        if vals.get('user_id') or vals.get('schedule_date'):
-            self.activity_update()
-        if self._need_new_activity(vals):
-            # need to change description of activity also so unlink old and create new activity
-            self.activity_unlink(['maintenance.mail_act_maintenance_request'])
-            self.activity_update()
+        if 'kanban_state' in vals:
+            done_requests = self.filtered(lambda m: m.kanban_state == 'done')
```

Comment:
```suggestion
            done_requests = self.filtered(lambda request: request.kanban_state == 'done')
```

Same for everywhere, where it's needed.